torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,215 @@
+    map[kDilation] = inputs[kConv2dDilation];
+    map[kGroups] = inputs[kConv2dGroups];
+  } else if (fname == kGemmOp) {
+    std::vector<int> tensors{0, 1};
+    bool check = validateInput(fname, 2, inputs, tensors);
+    if (!check) {
+      return map;
+    }
+
+    at::Tensor left = inputs[0].toTensor();
+    at::Tensor right = inputs[1].toTensor();
+    map[kMat1Size] = at::IValue(left.sizes());
+    map[kMat2Size] = at::IValue(right.sizes());
+  } else if (fname == kMulOp) {
+    std::vector<int> tensors{0};
+    bool check = validateInput(fname, 1, inputs, tensors);
+    if (!check) {
+      return map;
+    }
+
+    at::Tensor mat = inputs[0].toTensor();
+    map[kMatSize] = at::IValue(mat.sizes());
+  } else if (fname == kAddOp) {
+    std::vector<int> tensors{0};
+    bool check = validateInput(fname, 1, inputs, tensors);
+    if (!check) {
+      return map;
+    }
+
+    at::Tensor mat = inputs[0].toTensor();
+    map[kMatSize] = at::IValue(mat.sizes());
+  }
+
+  return map;
+}
+
+uint64_t computeFlops(const std::string &op_name, const std::unordered_map<std::string, c10::IValue> &extra_args) {
+  if (op_name == kConv2dOp) {
+    if (extra_args.find(kInputSize) == extra_args.end()
+        || extra_args.find(kWeightSize) == extra_args.end()) {
+      TORCH_WARN("Calculating flops for aten::conv2d requires input_size and weight_size in saved arguments.");
+      return 0;
+    }
+    auto input_sizes_ref = extra_args.at(kInputSize);
+    auto kernel_sizes_ref = extra_args.at(kWeightSize);
+    if (!input_sizes_ref.isIntList() || !kernel_sizes_ref.isIntList()) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires input and weight tensor sizes.");
+      return 0;
+    }
+
+    const std::vector<int64_t> input_sizes = input_sizes_ref.toIntVector();
+    const std::vector<int64_t> kernel_sizes = kernel_sizes_ref.toIntVector();
+    if (input_sizes.size() != 4 || kernel_sizes.size() != 4) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because both input and weight must be size 4.");
+      return 0;
+    }
+    // format of the input is defined in torch.nn.quantized.functional.conv2d()
+    uint64_t minibatch = 0, in_channels = 0, input_h = 0, input_w = 0;
+    uint64_t out_channels = 0, kernel_h = 0, kernel_w = 0;
+    const uint64_t conv2d_multiply_factor = 2;
+    std::tie(minibatch, in_channels, input_h, input_w) = std::make_tuple(input_sizes[0], input_sizes[1],
+                                                                         input_sizes[2], input_sizes[3]);
+    std::tie(out_channels, std::ignore, kernel_h, kernel_w) = std::make_tuple(kernel_sizes[0], kernel_sizes[1],
+                                                                              kernel_sizes[2], kernel_sizes[3]);
+
+    // grouping is NOT properly handled yet
+    return conv2d_multiply_factor * minibatch * input_h * input_w * kernel_h * kernel_w * in_channels * out_channels;
```

Comment:
To properly handle stride, pad, distillation, and so on, we should use output_h * output_w instead of input_h * input_w .